### PR TITLE
[#246] Added poly_rectify(style=strict|planarized)

### LINKED ISSUE #246: Define and implement cap-mode semantics for poly_rectify
Parent: #240

Related: #243, #245

## Goal

Bring `poly_rectify(...)` up to speed for irregular and valence > 3 source vertices, including cases such as rhombic dodecahedron where the current midpoint vertex faces can be non-planar or otherwise unsuitable.

## Core design issue

`poly_truncate(...)` can use two generated points per source edge, one near each endpoint. That lets each source vertex cap choose its own cap plane.

Classic rectification uses one generated point per source edge: the edge midpoint. A planar cap at one endpoint and a planar cap at the other endpoint may want different points on the same source edge. So `cap_mode` for rectification is not a simple parameter copy from truncation.

## Options to decide

1. **Strict rectification**
   Keep one midpoint per source edge. This preserves classic topology but cannot promise planar valence > 3 vertex faces.

2. **Planarized rectification**
   Use two generated points per source edge, one per endpoint, computed through vertex-figure realization. This is closer to `poly_truncate(..., t ~= 0.5)` with explicit topology, not pure midpoint rectification.

3. **Shared-edge solve**
   Choose one point per edge that best satisfies adjacent vertex cap planes. This is probably more complex and less transparent; treat as future work unless it is clearly needed.

## Proposed first implementation

Add an explicit mode/parameter that distinguishes strict midpoint rectification from planarized rectification, rather than silently changing `poly_rectify(...)` behavior.

Potential API direction:

```scad
poly_rectify(poly, cap_mode="edge_fraction", rectify_mode="strict" | "planarized", ...)
```

Final names can change, but the distinction should be visible to users.

## Tests

- Add a failing/current-behavior test for a rhombic dodecahedron or similar irregular valence > 3 source.
- Add tests proving strict rectification retains midpoint/shared-edge behavior.
- Add tests proving planarized rectification produces planar vertex faces for the selected stress case.
- Add visual regression examples comparing strict vs planarized behavior.

## Docs

Document clearly that planarized rectification is not identical to classic shared-midpoint rectification when edge endpoints require different cap planes.

### PR DESCRIPTION
**Pull request by susanw1**

### CODE REVIEW

**Comment by susanw1** [comment]
Changed:

  - Allowed `poly_truncate(..., t=0.5)` by removing the old rejection and documenting the midpoint meaning for cap_mode="edge_fraction".
  - Added `poly_rectify(style="strict"|"planarized", cap_mode=...)` in _src/polysymmetrica/core/truncation.scad:545_.
  - Kept `style="strict"` on the old lightweight shared-midpoint path in _src/polysymmetrica/core/truncation.scad:342_.
  - Added planarized per-vertex depth solving in _src/polysymmetrica/core/truncation.scad:305_.
  - Added focused tests in _src/tests/core/TestTruncation.scad:232_.
  - Added a new visual regression case and two baseline PNGs for strict vs planarized tetrakis rectification.
  - Updated the transform tutorial text.

File: src/tests/core/TestTruncation.scad [hunk 1]
```diff
@@ -214,6 +229,29 @@ module test_poly_rectify__tetra_counts() {
     assert(_count_faces_of_size(q,3) == 8, "rectify tetra: 8 triangles");
 }
 
+module test_poly_truncate__t_half_edge_fraction_matches_strict_rectify() {
+    p = octahedron();
+    q = poly_truncate(p, t = 0.5, cap_mode = "edge_fraction");
```
[the remark this answers is not in the document]
Comment (susanw1) [reply src/tests/core/TestTruncation.scad:234]: Fixed.
@codex review again

File: src/polysymmetrica/core/truncation.scad [hunk 2]
```diff
@@ -360,7 +466,6 @@ function poly_truncate(
                     is_undef(mode_ov) ? cap_mode : mode_ov
             ],
             _cap_modes_ok = _ps_truncate_cap_modes_ok(cap_mode_by_vert),
```
[the remark this answers is not in the document]
Comment (susanw1) [reply src/polysymmetrica/core/truncation.scad:495]: Yes, that review point is right. I fixed it by adding a realized edge-fraction overlap guard in src/polysymmetrica/core/truncation.scad:297. It asserts when both endpoint cuts are intended to be within the half-edge regime but the realized cap-mode intersections cross on the edge.

@codex review again
